### PR TITLE
[SCv2] Improve inference docs re routing and headers

### PR DESCRIPTION
**Pull request by agrski**
**What this PR does / why we need it**:
This PR achieves two things.  Firstly, it adds information on routing _to_ Core v2 via different methods (authority/virtual host headers, path matching, etc.).  Secondly, it improves the existing documentation on routing _through_ Core v2 with models and pipelines.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/source/contents/inference/index.md
Comment (sakoush) [inline docs/source/contents/inference/index.md:8]: do we provide examples how we can write directly to a kafka topic?

File: docs/source/contents/inference/index.md
Comment (sakoush) [inline docs/source/contents/inference/index.md:52]: can we provide a link for more details on Istio / Ambassador setup?

File: docs/source/contents/inference/index.md
Comment (sakoush) [inline docs/source/contents/inference/index.md:63]: what will happen if the pipeline does not have an output section? 503?

File: docs/source/contents/inference/index.md
Comment (sakoush) [inline docs/source/contents/inference/index.md:59]: explain 1`iris`  schema as it is used in the example inferences below?

File: docs/source/contents/inference/index.md
Comment (sakoush) [inline docs/source/contents/inference/index.md:344]: perhaps we want an example as well for how to write to kafka?

**Review by sakoush — APPROVED** [review]
Lovely. Left some minor comments.

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *[sakoush](https://github.com/SeldonIO/seldon-core/pull/4481#pullrequestreview-1215238832 "Approved")*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/v2/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: docs/source/contents/inference/index.md
Comment (agrski) [reply docs/source/contents/inference/index.md:8]: I don't think we have any such examples anywhere currently, at least not proper ones.
It's probably one to add an issue for/add to the roadmap.

There might be some under `scheduler/data-flow/scripts`, but they'd need to be checked as still being valid and they're not in Python, so less notebook friendly.

File: docs/source/contents/inference/index.md
Comment (agrski) [reply docs/source/contents/inference/index.md:52]: We can do.

My thinking was to not add too many links on generic subjects like this, in case it becomes a matter of needing to add links to all major service meshes and for lots of different sections.
In turn, we should then probably add links in other sections of this doc for ingress routing, etc.
Basically, I'd rather put the onus on readers to check the docs for their preferred tools rather than us having to maintain a list of these.

Do you think that's fair, or would you indeed prefer links to at least the mentioned ones for the various sections?

File: docs/source/contents/inference/index.md
Comment (agrski) [reply docs/source/contents/inference/index.md:63]: That's a good question!  I'm not sure off-hand what the answer is.  I'd probably expect a _timeout_ from the pipeline gateway, but would need to check.

Are you happy for this to be defined in a follow-up PR?

File: docs/source/contents/inference/index.md
Comment (agrski) [reply docs/source/contents/inference/index.md:59]: By `schema` do you mean the data schema or the model definition/k8s manifest?

File: docs/source/contents/inference/index.md
Comment (sakoush) [reply docs/source/contents/inference/index.md:52]: If this is not seldon specific then yes it makes sense to keep it as is.

File: docs/source/contents/inference/index.md
Comment (sakoush) [reply docs/source/contents/inference/index.md:8]: A follow up ticket if not easily available makes sense.

File: docs/source/contents/inference/index.md
Comment (sakoush) [reply docs/source/contents/inference/index.md:59]: I meant the data schema as the inference payload uses it and perhaps it is not very clearly why the payload is constructed as such for new users coming of v2 protocol?

Comment (sakoush) [reply docs/source/contents/inference/index.md:59]: I meant the data schema as the inference payload uses it and perhaps it is not very clearly why the payload is constructed as such for new users coming of v2 protocol?

File: docs/source/contents/inference/index.md
Comment (sakoush) [reply docs/source/contents/inference/index.md:63]: sure

File: docs/source/contents/inference/index.md
Comment (agrski) [reply docs/source/contents/inference/index.md:52]: Indeed - not Seldon-specific

File: docs/source/contents/inference/index.md
Comment (agrski) [reply docs/source/contents/inference/index.md:59]: Fixed in https://github.com/SeldonIO/seldon-core/pull/4481/commits/685ca37821f0e1da416fa40fdc9cc3b10a51383d

**Pull request merged**
